it/src/main/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManager.java

```diff
@@ -0,0 +1,219 @@
+ *
+ * <p>There are one instance, one database, and multiple tables per manager object. The tables are
+ * under the database, and the database is under the instance. Instance creation is handled by
+ * {@link Builder#build()} using {@link DefaultSpannerResourceManager#createInstance(String,
+ * String)}. Database creation is triggered by the {@link
+ * DefaultSpannerResourceManager#createTable(String)} method, if there is no existing database.
+ *
+ * <p>The instance and database ids are formed using testId. The database id string will be
+ * {testId}, with some extra formatting. The instance id will be "{testId}-{ISO8601 time,
+ * microsecond precision}", with additional formatting.
+ *
+ * <p>Note: If testId is more than 30 characters, a new testId will be formed for naming: {first 22
+ * chars of long testId} + “-” + {7 char hash of testId}.
+ */
+public class DefaultSpannerResourceManager implements SpannerResourceManager {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultSpannerResourceManager.class);
+
+  private final InstanceId instanceId;
+  private final DatabaseId databaseId;
+
+  private final Spanner spanner;
+  private final InstanceAdminClient instanceAdminClient;
+  private final DatabaseAdminClient databaseAdminClient;
+  private DatabaseClient databaseClient;
+
+  private DefaultSpannerResourceManager(Builder builder) {
+    this(builder.projectId, builder.instanceId, builder.databaseId);
+  }
+
+  @VisibleForTesting
+  DefaultSpannerResourceManager(String projectId, String instanceId, String databaseId) {
+    this.instanceId = InstanceId.of(projectId, instanceId);
+    this.databaseId = DatabaseId.of(projectId, instanceId, databaseId);
+    this.spanner = SpannerOptions.newBuilder().setProjectId(projectId).build().getService();
+    this.instanceAdminClient = spanner.getInstanceAdminClient();
+    this.databaseAdminClient = spanner.getDatabaseAdminClient();
+  }
+
+  public static Builder builder(String testId, String projectId, String region) {
+    return new Builder(testId, projectId, region);
+  }
+
+  @VisibleForTesting
+  synchronized void createInstance(String projectId, String region) throws CreateInstanceException {
+    InstanceInfo instanceInfo =
+        InstanceInfo.newBuilder(instanceId)
+            .setInstanceConfigId(InstanceConfigId.of(projectId, region))
+            .setDisplayName(instanceId.getInstance())
+            .setNodeCount(1)
+            .build();
+    try {
+      instanceAdminClient.createInstance(instanceInfo).get();
+      LOG.info("Successfully created instance.");
+    } catch (ExecutionException | InterruptedException e) {
+      throw new CreateInstanceException("Failed to create instance.", e);
+    }
+  }
+
+  @Override
+  public synchronized void createTable(String statement)
+      throws CreateDatabaseException, IllegalStateException, UpdateDatabaseException {
+    if (spanner.isClosed()) {
```

Comment:
This check is being reused a lot and with an identical error message. It would be better to have a private method that does the check and throws the exception.